sw/device/lib/runtime/heap.cpp

```diff
@@ -0,0 +1,60 @@
+/* Additional heap functions required for C++
+ * 
```

Comment:
can you please adapt the header to the ones we are using with your name?

or copying it from here https://www.apache.org/licenses/LICENSE-2.0.html

```
Copyright [yyyy] [name of copyright owner]

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
```